packages/router/__tests__/router-test.ts

```diff
@@ -1825,12 +1833,12 @@ describe("a router", () => {
       expect(rootLoader.mock.calls.length).toBe(1);
       expect(shouldRevalidate.mock.calls[0][0]).toMatchObject({
         currentParams: {},
-        currentUrl: new URL("http://localhost/child"),
+        currentUrl: expect.URL("http://localhost/child"),
```

Comment:
Found out that `new URL` inside `toMatchObject` doesn't actually validate the internal URL 🤦 , so added the custom matcher and changed all existing instances.  Thankfully no false positives.